custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -95,64 +104,92 @@ def allowed_to_forward(self, episode_case):
-            voucher_count = episode_case_properties.get('approved_voucher_count', 0)
-            if voucher_count < 2:
-                voucher_count = len(
-                    get_approved_prescription_vouchers_from_episode(episode.domain, episode.case_id)
-                )
-            return voucher_count
+    @staticmethod
+    def _property_as_date(properties, property_name):
+        """
+        Parse the given value of the given property_name in the given property dict as a date.
+        If the property is not a date, return None
+        """
+        try:
+            return parse(properties.get(property_name, "nope"))
+        except ValueError:
+            return None
+
+    @staticmethod
+    def prescription_total_days_threshold_in_trigger_state(episode_case_properties, n):
+        threshold_case_prop = BETSDrugRefillRepeater._get_threshold_case_prop(n)
+        return bool(
+            BETSDrugRefillRepeater._property_as_date(episode_case_properties, threshold_case_prop)
+            and episode_case_properties.get("event_{}_{}".format(DRUG_REFILL_EVENT, n)) != "sent"
+        )
 
-        not_sent = voucher_case_properties.get("event_{}".format(DRUG_REFILL_EVENT)) != "sent"
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        episode_case_properties = episode_case.dynamic_case_properties()
+        trigger_by_threshold = {}  # threshold -> boolean
+        threshold_prop_values_by_threshold = {}  # threshold -> date
+
+        for n in TOTAL_DAY_THRESHOLDS:
+            threshold_case_prop = self._get_threshold_case_prop(n)
+            threshold_prop_values_by_threshold[n] = self._property_as_date(
+                episode_case_properties, threshold_case_prop
+            )
+            trigger_for_n = bool(
+                self.prescription_total_days_threshold_in_trigger_state(episode_case_properties, n)
+                and case_properties_changed(episode_case, [threshold_case_prop])
+            )
+            trigger_by_threshold[n] = trigger_for_n
+
+        trigger_dates_unique = self._list_items_unique(filter(None, threshold_prop_values_by_threshold.values()))
+        if not trigger_dates_unique:
+            self._flag_program_team()
 
         return (
-            # TODO: Confirm state == "fulfilled"
-            voucher_case_properties.get("state") == "fulfilled"
-            and voucher_case_properties.get("type") == "prescription"
-            and not_sent
-            and case_properties_changed(voucher_case, ['state'])
-            and is_valid_voucher_submission(voucher_case)
-            and _get_voucher_count() >= 2
+            trigger_dates_unique
+            and any(trigger_by_threshold.values())
+            and is_valid_episode_submission(episode_case)
         )
 
+    def _flag_program_team(self):
+        # TODO: Write me
```

Comment:
Can't implement until we hear who should be notified, fine to merge before this is determined.